server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1033,6 +1042,34 @@ public void parse(DocumentParserContext context) throws IOException {
         if (Token.VALUE_NULL == context.parser().currentToken()) {
             return;
         }
+        if (fieldType().dims == null) {
+            int dims = elementType.parseDimensionCount(context);
+            String fieldTypeName = (context.parent().fullPath().equals("_doc") == false) ? simpleName() : name();
+            DenseVectorFieldType updatedDenseVectorFieldType = new DenseVectorFieldType(
+                fieldTypeName,
+                indexCreatedVersion,
+                elementType,
+                dims,
+                indexed,
+                similarity,
+                fieldType().meta()
+            );
+            Mapper update = new DenseVectorFieldMapper(
+                name(),
+                updatedDenseVectorFieldType,
+                elementType,
+                dims,
+                indexed,
+                similarity,
+                indexOptions,
+                indexCreatedVersion,
+                multiFields(),
+                copyTo
+            );
+            context.addDynamicMapper(update);
+
+            return;
+        }
```

Comment:
```suggestion
            DenseVectorFieldType updatedDenseVectorFieldType = new DenseVectorFieldType(
                fieldType().name(),
                indexCreatedVersion,
                elementType,
                dims,
                indexed,
                similarity,
                fieldType().meta()
            );
            Mapper update = new DenseVectorFieldMapper(
                simpleName(),
                updatedDenseVectorFieldType,
                elementType,
                dims,
                indexed,
                similarity,
                indexOptions,
                indexCreatedVersion,
                multiFields(),
                copyTo
            );
            context.addDynamicMapper(update);
            return;
        }
```

This is what it should be I think. I made this change locally and all the yaml tests passed. What do you think?

Reply:
That works for me, will update - thanks!